iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -137,99 +154,440 @@ public static RegistryManager getRegistyManager(String iotHubConnectionString) t
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing device {} into testSasDeviceWithTwinQueue", testIdentity.getDeviceId());
+                        testSasDeviceWithTwinQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing device {} into testSasDeviceQueue", testIdentity.getDeviceId());
+                        testSasDeviceQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                }
+                else
+                {
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing device {} into testX509DeviceWithTwinQueue", testIdentity.getDeviceId());
+                        testX509DeviceWithTwinQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing device {} into testX509DeviceQueue", testIdentity.getDeviceId());
+                        testX509DeviceQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                }
+            }
+            else if (testIdentity instanceof TestModuleIdentity)
+            {
+                if (((TestModuleIdentity) testIdentity).getModule().getAuthenticationType() == AuthenticationType.SAS)
+                {
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing module {} into testSasModuleWithTwinQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testSasModuleWithTwinQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing module {} into testSasModuleQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testSasModuleQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                }
+                else
+                {
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing module {} into testX509ModuleWithTwinQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testX509ModuleWithTwinQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing module {} into testX509ModuleQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testX509ModuleQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void disposeTestDevices(Iterable<Device> devices, String iotHubConnectionString)
+    {
+        List<Device> subList = new ArrayList<>();
```

Comment:
It would be good to print out the total devices being disposed just to know how many we created for the tests.